src/OpenFOAM/containers/Lists/gpuList/textures.H

```diff
@@ -31,18 +34,25 @@ public:
 
     void destroy()
     {
+#if __CUDA_ARCH__ >= 300
         cudaDestroyTextureObject(tex);
+#else
+        delete tex;
+#endif
     }
 };
 
```

Comment:
Maybe you could reduce the number of '#ifs' by putting all definitions of 3.0 functions under one 'if' and the 2.0 functions under 'else'. The 2.0 operator [] does not have be specialized for any of the types since for every type the function looks the same.